server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -1015,11 +1015,12 @@ public UserVm upgradeVirtualMachine(UpgradeVMCmd cmd) throws ResourceAllocationE
         int currentCpu = currentServiceOffering.getCpu();
         int currentMemory = currentServiceOffering.getRamSize();
 
+        Account owner = _accountMgr.getActiveAccountById(vmInstance.getAccountId());
         if (newCpu > currentCpu) {
-            _resourceLimitMgr.checkResourceLimit(caller, ResourceType.cpu, newCpu - currentCpu);
+            _resourceLimitMgr.checkResourceLimit(owner, ResourceType.cpu, newCpu - currentCpu);
         }
         if (newMemory > currentMemory) {
-            _resourceLimitMgr.checkResourceLimit(caller, ResourceType.memory, newMemory - currentMemory);
+            _resourceLimitMgr.checkResourceLimit(owner, ResourceType.memory, newMemory - currentMemory);
         }
 
         // Check that the specified service offering ID is valid
```

Comment:
I previously saw code like this in https://github.com/apache/cloudstack/pull/3759/files. Does this belong into this PR?

Reply:
@DennisKonrad I do not think so.
@bwsw could you use "git rebase" with latest master, instead of "git merge" ?

Reply:
@weizhouapache @DennisKonrad resolved.